apps/epl_ets/src/epl_ets_metric.erl

```diff
@@ -0,0 +1,186 @@
+    {Tab, InfoMap}.
+
+transform_traces_by_pid(Traces) ->
+    TracesByPid = split_traces_by_pid(Traces),
+    TracesSortedByMs = sort_traces_by_ms(TracesByPid),
+    TracesCleaned = clean_traces_unpaired(TracesSortedByMs),
+    TracesCallTime = calculate_calls_time(TracesCleaned),
+    lists:merge(TracesCallTime).
+
+transform_traces_by_tab(Traces) ->
+    TracesByTab = split_traces_by_tab(Traces),
+    format_traces(TracesByTab).
+
+calculate_tabs_statistics(Stats) ->
+    [calculate_tab_statistics(Stat) || Stat <- Stats].
+
+calculate_tab_statistics({Tab, FuncStats}) ->
+    {Tab, [calculate_tab_func_statistics(FuncStat) || FuncStat <- FuncStats]}.
+    
+calculate_tab_func_statistics({Func, TimeProbes}) ->
+    #{<<"func">> => namify(Func), <<"max_time">> => max(TimeProbes), 
+      <<"count">> => erlang:length(TimeProbes)}.
+
+split_traces_by_pid(Traces) ->
+    split_traces_by(Traces, 1).
+
+split_traces_by_tab(Traces) ->
+    split_traces_by(Traces, 2).
+
+split_traces_by_func(Traces) ->
+    split_traces_by(Traces, 3).
+
+split_traces_by(Traces, ElemIndex) ->
+    TracesUnique = lists:ukeysort(ElemIndex, Traces),
+    KeysUnique = lists:map(fun(T) -> element(ElemIndex, T) end, TracesUnique),
+    Splitted = [lists:partition(fun(T) -> element(ElemIndex, T) =:= KU end,
+                                Traces) || KU <- KeysUnique],
+    lists:map(fun({S, _NS}) -> S end, Splitted).
+
+format_traces(TraceLists) ->
+    [format_trace(T) || T <- TraceLists].
+
+format_trace(Traces = [{_Pid, Tab, _Func, _T} | _]) ->
+    TracesByFunc = split_traces_by_func(Traces),
+    {Tab, [format_trace_by_func(T) || T <- TracesByFunc]}.
+
+format_trace_by_func(Traces = [{_Pid, _Tab, Func, _T} | _]) ->
+    {Func, lists:map(fun({_, _, _, MS}) -> trunc_float(MS, 4) end, Traces)}.
+
+clean_traces_unpaired(TraceLists) ->
+    TL = [clean_first_trace(T) || T <- TraceLists],
+    TLCleared = [clean_last_trace(T2, get_last_elem(T2)) || T2 <- TL],
+    [lists:filter(fun(E) -> E =/= [] end, T3) || T3 <- TLCleared].
+
+clean_first_trace([{_Pid, null, return_to, _TS} | Traces]) ->
+    Traces;
+clean_first_trace(Traces) ->
+    Traces.
+
+clean_last_trace(Traces, null) ->
+    Traces;
+clean_last_trace(Traces, {_Pid, null, return_to, _TS}) ->
+    Traces;
+clean_last_trace(Traces, _LT) ->
+    lists:droplast(Traces).
+
+sort_traces_by_ms(TraceLists) ->
+    TraceListsMs = [trace_ts_to_ms(TL) || TL <- TraceLists],
+    [sort_trace_by_ms(TLMs) || TLMs <- TraceListsMs].
+
+trace_ts_to_ms(Traces) ->
+    lists:map(fun({Pid, T, F, TS}) -> {Pid, T, F, ts_to_ms(TS)} end, Traces).
+
+ts_to_ms({Mega, Sec, Micro}) ->
+    (Mega*1000000 + Sec)*1000 + Micro/1000.
+
+sort_trace_by_ms(Traces) ->
+    lists:sort(fun({_, _, _, M1}, {_, _, _, M2}) -> M1 < M2 end, Traces).
+
+calculate_calls_time(TraceLists) ->
+    [calculate_call_time(T) || T <- TraceLists].
+
+calculate_call_time(Traces) ->
+    PairStartPoint = lists:seq(1, erlang:length(Traces) - 1, 2),
```

Comment:
The numbers should be kept as variables or returned by fucntions so that one doesn't have to guess the meaning.